src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+        c.argument('description', help='Multi-line explain this resource.')
+        c.argument('target_scope', arg_type=get_enum_type(['subscription', 'managementGroup']), help='The scope where this blueprint definition can be assigned.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters required by this blueprint definition.')
+
+    with self.argument_context('blueprint import') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('input_path', help='The directory path for json definitions of blueprint and artifacts. Artifacts json files should be in input-path/artifacts/.', completer=FilesCompleter())
+
+    with self.argument_context('blueprint update') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('description', help='Multi-line explain this resource.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters required by this blueprint definition.')
+
+    with self.argument_context('blueprint delete') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint show') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint list') as c:
+        pass
+
+    with self.argument_context('blueprint artifact delete') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+
+    with self.argument_context('blueprint artifact show') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('artifact_name', help='Name of the blueprint artifact.')
+
+    with self.argument_context('blueprint artifact list') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+
+    with self.argument_context('blueprint resource-group create') as c:
+        c.argument('blueprint_name', help='Name of the blueprint definition.')
+        c.argument('rg_name', help='Name of this resource group. Leave empty if the resource group name will be specified during the blueprint assignment.')
+        c.argument('rg_location', help='Location of this resource group. Leave empty if the resource group location will be specified during the blueprint assignment.')
+        c.argument('artifact_name', help='A unique name of this resource group artifact.')
+        c.argument('display_name', help='Display name of this resource group artifact.')
+        c.argument('description', help='Description of the blueprint artifact.')
+        c.argument('depends_on', nargs='+', help='Artifacts which need to be deployed before the specified artifact.')

```

Comment:
the parameter name is a little strange. what about dependencies?

Reply:
This is from the swagger. I think for arguments we can use verbs like `enable-xx`, so `depends-on` should be ok?